### PR TITLE
symengine/pow.cpp: Print `exp(1)` instead of `E` to maintain consistency

### PR DESCRIPTION
**Pull request by prateeksingh0001**
Closes https://github.com/symengine/symengine/issues/1225

### CODE REVIEW

**Comment by ranjithkumar007** [comment]
@prateeksingh0001, Can you add some tests ?

**Comment by prateeksingh0001** [comment]
@isuruf after your review I looked into the issue again and I was able to solve the issue of printing only when the exp() function is being used inside an operation but not when it is being used separately, I did this by adding an if condition in printer.cpp. For eg:
`mul(exp(integer(1)), integer(2))` is being printed as `2**exp(1)` but simply `exp(integer(1))` is printed as `E` only.
Kindly help me with some pointers as to how I can solve the above issue.

**Comment by isuruf** [comment]
@prateeksingh000, have you tried debugging the code to see how `exp(1)` is printed?

File: symengine/printer.cpp [hunk 1]
```diff
@@ -620,7 +623,16 @@ void StrPrinter::bvisit(const Log &x)
...
 void StrPrinter::bvisit(const Constant &x)
 {
-    str_ = x.get_name();
+    /*If the constant is E then it is to
+      be printed as exp(1), this is done
+      to maintain consistency with sympy*/
+    if(x.get_name() == "E") {
+        std::ostringstream s;
+        s << "exp(1)";
+        str_ = s.str() ;
```
Comment (isuruf) [inline symengine/printer.cpp]: `str_ = "exp(1)"` is enough.

File: symengine/printer.cpp [hunk 2]
```diff
@@ -620,7 +623,16 @@ void StrPrinter::bvisit(const Log &x)
 
 void StrPrinter::bvisit(const Constant &x)
 {
-    str_ = x.get_name();
+    /*If the constant is E then it is to
+      be printed as exp(1), this is done
+      to maintain consistency with sympy*/
```
Comment (isuruf) [inline symengine/printer.cpp]: This is done to maintain consistency, not for sympy. SymPy prints it as `E`

File: symengine/printer.cpp [hunk 3]
```diff
@@ -379,7 +379,10 @@ void StrPrinter::bvisit(const Mul &x)
             den++;
         } else {
             if (eq(*(p.second), *one)) {
```
Comment (isuruf) [inline symengine/printer.cpp]: Change this to `eq(*(p.second), *one) and neq(*(p.first), *E)` and you won't need the changes below

File: symengine/printer.cpp [hunk 4]
```diff
@@ -620,7 +623,16 @@ void StrPrinter::bvisit(const Log &x)
 
 void StrPrinter::bvisit(const Constant &x)
 {
-    str_ = x.get_name();
+    /*If the constant is E then it is to
+      be printed as exp(1), this is done
+      to maintain consistency with sympy*/
+    if(x.get_name() == "E") {
```
Comment (isuruf) [inline symengine/printer.cpp]: Format using clang-format. See https://github.com/symengine/symengine/wiki/Development-workflow-for-SymEngine#committing-code-to-be-done-before-each-commit

File: symengine/printer.cpp (hunk 2 above)
Comment (isuruf) [inline symengine/printer.cpp]: ping

Comment (prateeksingh0001) [inline symengine/printer.cpp]: Understood, I've made the changes accordingly :smile:

**Comment by certik** [comment]
Why did this test fail? https://travis-ci.org/symengine/symengine/jobs/214969418

**Comment by certik** [comment]
Regarding formatting, I created a new issue for this #1238, as it's the second time I got confused by that.
I am glad we do that though, as I was wondering that the comment looks ugly.

**Comment by ranjithkumar007** [comment]
Looks good,+1.

**Comment by isuruf** [comment]
@prateeksingh0001, thanks for the PR. Let's discuss whether we really should do this or not in the issue.

**Comment by prateeksingh0001** [comment]
Sure,  we should definitely discuss this first.

**Comment by isuruf** [comment]
Thanks, @prateeksingh0001 for the PR. We are going to go with `E` as the default and have different printers for different languages

**Comment by prateeksingh0001** [comment]
No problem :)

**Pull request closed without merging**